src/utils/statuses/groupStatusesByUmbrellaStatus.ts

```diff
@@ -26,14 +27,15 @@ export const groupStatusesByUmbrellaStatus = (
     const umbrellaStatus = mapTaskStatusToUmbrellaStatus[stat.status];
     if (counts[umbrellaStatus]) {
       counts[umbrellaStatus].count += stat.count;
-      if (!counts[umbrellaStatus].statuses.includes(stat.status)) {
-        counts[umbrellaStatus].statuses.push(taskStatusToCopy[stat.status]);
-      }
+      counts[umbrellaStatus].statuses = deduplicatedAppend(
```

Comment:
Would making `ColorCount.statuses` a Set of strings eliminate the need for this method?

Reply:
Discussed offline but commenting for visibility. The implementations of this method require an array so the implementers would need to cast the set into an array in order to use this field. Which makes implementing this method more complex. We opted instead to use a set within `deduplicatedAppend`  and have it return a deduped array instead.